tests/python/unittest/test_optimizer.py

```diff
@@ -196,14 +211,24 @@ def update(self, index, weight, grad, state):
         lr = self._get_lr(index)
         wd = self._get_wd(index)
         self._update_count(index)
-        n, g, delta = state
         grad = grad * self.rescale_grad
```

Comment:
As described in https://github.com/dmlc/mxnet/issues/5099, the old way of handling weight decay is not correct. We may need to use `grad = grad * self.rescale_grad + wd * weight`.